### PR TITLE
Container examples with traefik

### LINKED ISSUE #1500: Create container examples using Traefik instead of nginx
### Plone Documentation versions affected

- [ ] 5.x
- [X] 6.x

### Description

For each example in https://6.docs.plone.org/install/containers/examples/index.html, we should have a similar one using Traefik

### PR DESCRIPTION
**Pull request by margaridasp**
Fixes #1500 

I think we still need to create simpler examples with less components.

### CODE REVIEW

**Review by ericof — APPROVED** [review]

**Comment by ericof** [comment]
@margaridasp Let's merge this one, then create other examples :-)

File: docs/glossary.md [hunk 1]
```diff
@@ -384,6 +384,9 @@ HAProxy
 nginx
     [nginx](https://docs.nginx.com/nginx/) (pronounced "engine x") is an HTTP and reverse proxy server, a mail proxy server, and a generic TCP/UDP proxy server, originally written by Igor Sysoev.
 
+traefik
```
Comment (stevepiercy) [inline docs/glossary.md]: This is a proper name and there are many products under the Traefik brand name. We should be explicit to disambiguate it from all their other products.

`Traefik` should also be added to https://github.com/plone/documentation/blob/6.0/styles/Vocab/Plone/accept.txt

```suggestion
Traefik Proxy
```

File: docs/glossary.md [hunk 2]
```diff
@@ -384,6 +384,9 @@ HAProxy
...
+    [traefik](https://traefik.io/) is an open-source reverse proxy and load balancer, suitable for containerized architectures.
```
Comment (stevepiercy) [inline docs/glossary.md]:
```suggestion
    [Traefik Proxy](https://traefik.io/traefik/) is an open-source reverse proxy and load balancer, suitable for containerized architectures.
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 3]
```diff
@@ -0,0 +1,478 @@
+---
+myst:
+  html_meta:
+    "description": "Very simple Plone 6 setup with only one backend and data being persisted in a Docker volume."
+    "property=og:description": "Very simple Plone 6 setup with only one backend and data being persisted in a Docker volume."
+    "property=og:title": "traefik, Frontend, Backend, Varnish container example"
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]:
```suggestion
    "property=og:title": "Traefik Proxy, Frontend, Backend, Varnish container example"
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+    "keywords": "Plone 6, Container, Docker, traefik, Frontend, Backend, Varnish"
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]:
```suggestion
    "keywords": "Plone 6, Container, Docker, Traefik Proxy, Frontend, Backend, Varnish"
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 5]
```diff
@@ -0,0 +1,478 @@
...
+---
+
+# traefik, Frontend, Backend, Varnish container example
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]:
```suggestion
# Traefik Proxy, Frontend, Backend, Varnish container example
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 6]
```diff
@@ -0,0 +1,478 @@
...
+
+This example is a very simple setup with one backend and data being persisted in a Docker volume.
+
+{term}`traefik` in this example is used as a reverse proxy.
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]:
```suggestion
{term}`Traefik Proxy` in this example is used as a reverse proxy.
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 7]
```diff
@@ -0,0 +1,478 @@
...
+```
+
+Change into your project directory.
+
+```shell
+cd traefik-volto-plone-varnish
+```
+
+
+### Varnish configuration
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]: Let's promote it one level.

```suggestion
## Varnish configuration
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 8]
```diff
@@ -0,0 +1,478 @@
...
+
+This example is a very simple setup with one backend and data being persisted in a Docker volume.
+
+{term}`traefik` in this example is used as a reverse proxy.
+
+{term}`Varnish` is used for caching.
+
+A purger component is also used. This solves the problem of invalidating the cache in multiple Varnish servers, which could be desirable in containerized deployment.
+
+## Setup
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]: This is not really "setup".

```suggestion
## Create a project space
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 9]
```diff
@@ -0,0 +1,478 @@
...
+  }
+}
+```
+
+```{note}
+`http://plone.localhost/` is the URL you will be using to access the website.
+You can either use `localhost`, or add it in your `/etc/hosts` file or DNS to point to the Docker host IP.
+```
+
+### Service configuration with Docker Compose
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]: Let's promote this one level, too.

```suggestion
## Service configuration with Docker Compose
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 10]
```diff
@@ -0,0 +1,478 @@
...
+
+Now let's create a `docker-compose.yml` file:
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]:
```suggestion
Now let's create a {file}`docker-compose.yml` file:
```

File: docs/install/containers/examples/traefik-volto-plone-varnish.md [hunk 11]
```diff
@@ -0,0 +1,478 @@
...
+
+### Varnish configuration
+
+Create an empty directory named `etc`.
+
+```shell
+mkdir etc
+```
+
+Add there a `varnish.vcl` that will be used by the varnish image:
```
Comment (stevepiercy) [inline docs/install/containers/examples/traefik-volto-plone-varnish.md]:
```suggestion
Add there a file {file}`varnish.vcl` that will be used by the Varnish image:
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Minor suggestions for proper names, renaming headings, MyST syntax for files.

**Comment by stevepiercy** [comment]
It is safe to ignore the failing docs build. I have another PR open that fixes the redirect in https://github.com/plone/documentation/pull/1504

**Comment by ericof** [comment]
@stevepiercy We can merge this whenever you are ready :-)

**Review by stevepiercy — APPROVED** [review]
Thank you! I love me some tasty docs!

**Pull request merged**